plugins/notifications-backend-module-teams/src/processor/NotificationsTeamsProcessor.ts

```diff
@@ -0,0 +1,287 @@
+        await this.sendWebhookMessages(webhooks, webhookOptions);
+      }
+    } catch (e) {
+      this.logger.error(`Failed to send teams notification: ${e}`);
+    }
+
+    return options;
+  }
+
+  private createTeamsCard(options: NotificationSendOptions): TeamsWebhookCard {
+    return {
+      '@type': 'MessageCard',
+      '@context': 'http://schema.org/extensions',
+      title: options.payload.title,
+      summary: options.payload.description,
+      potentialAction: [
+        {
+          '@type': 'OpenUri',
+          name: 'See more',
+          targets: [
+            {
+              os: 'default',
+              uri: getAbsoluteNotificationLink(this.config, options),
+            },
+          ],
+        },
+      ],
+    };
+  }
+
+  async getEntitiesWebhooks(entityRefs: string[]): Promise<string[]> {
+    const toFetch: string[] = [];
+    const webhooks: string[] = [];
+    for (const entityRef of entityRefs) {
+      const webhook = await this.cache.get<string>(entityRef);
+      if (webhook) {
+        webhooks.push(webhook);
+      } else {
+        toFetch.push(entityRef);
+      }
+    }
+
+    const entities = await this.getEntities(toFetch);
+
+    for (const entity of entities) {
+      const webhook = entity?.metadata.annotations?.[ANNOTATION_TEAMS_WEBHOOK];
+      if (entity && webhook) {
+        webhooks.push(webhook);
+        await this.cache.set(stringifyEntityRef(entity), webhook, {
+          ttl: this.cacheTtl,
+        });
+      }
+    }
+    return webhooks;
+  }
+
+  async getEntities(entityRefs: string[]): Promise<(Entity | undefined)[]> {
+    const { token } = await this.auth.getPluginRequestToken({
+      onBehalfOf: await this.auth.getOwnServiceCredentials(),
+      targetPluginId: 'catalog',
+    });
+
+    const response = await this.catalogClient.getEntitiesByRefs(
+      {
+        entityRefs,
+        filter: {
+          [`metadata.annotations.${ANNOTATION_TEAMS_WEBHOOK}`]:
+            CATALOG_FILTER_EXISTS,
+        },
+        fields: [
+          'kind',
+          'metadata.name',
+          'metadata.namespace',
+          `metadata.annotations.${ANNOTATION_TEAMS_WEBHOOK}`,
+        ],
+      },
+      {
+        token,
+      },
+    );
+
+    return response.items;
+  }
+
+  private async sendWebhookMessages(
+    webhooks: string[],
+    options: TeamsWebhookCard,
+  ): Promise<void> {
+    if (webhooks.length === 0) {
+      return;
+    }
+
+    const throttle = pThrottle({
+      limit: this.concurrencyLimit,
+      interval: this.throttleInterval,
+    });
+    const throttled = throttle((hook: string, card: TeamsWebhookCard) =>
+      this.sendWebhookMessage(hook, card),
+    );
+    await Promise.all(webhooks.map(webhook => throttled(webhook, options)));
+  }
+
+  private async sendWebhookMessage(
+    webhook: string,
+    card: TeamsWebhookCard,
+  ): Promise<void> {
+    const response = await fetch(webhook, {
```

Comment:
[always use](https://github.com/backstage/backstage/blob/master/docs/architecture-decisions/adr013-use-node-fetch.md) the `node-fetch` version of fetch on the backend